src/Integration/State/TransferableVisualState.cs

```diff
@@ -0,0 +1,83 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TransferableVisualState.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Integration.TeamExplorer;
+using SonarLint.VisualStudio.Integration.WPF;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+
+namespace SonarLint.VisualStudio.Integration.State
+{
+    internal class TransferableVisualState : ViewModelBase
+    {
+        private readonly ObservableCollection<ServerViewModel> connectedServers = new ObservableCollection<ServerViewModel>();
+        private ProjectViewModel boundProject;
+        private bool isBusy;
+
+        public ObservableCollection<ServerViewModel> ConnectedServers
+        {
+            get
+            {
+                Debug.Assert(ThreadHelper.CheckAccess(), $"{nameof(ConnectedServers)} should only be accessed from the UI thread");
+                return this.connectedServers;
+            }
+        }
+
+        public bool HasBoundProject
```

Comment:
Previously we had an observable collection in order to get free notifications when the bound projects have changed. This was one of the compromises we had because we had this level of refactoring to address it (the requirement came late and wasn't taken into account originally).
Moving to a proper state (which is a VM as well) simplifies things and we're now able to expose this property instead of the previous work around.